api4/post.go

```diff
@@ -125,7 +126,7 @@ func createEphemeralPost(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	w.WriteHeader(http.StatusCreated)
 	rp = model.AddPostActionCookies(rp, c.App.PostActionCookieSecret())
-	rp = c.App.PreparePostForClient(rp, true, false)
+	rp = c.App.PreparePostForClient(rp, true, false, c.AppContext.Session().UserId)
```

Comment:
Good question, there are at least 3 user ids in scope:
* `c.AppContext.Session().UserId`
* `ephRequest.Post.UserId`
* `ephRequest.UserID`
I think the session id is the correct one to use because it's not given as a user input; we don't want users to be able to provide a user ID and then have access to posts that they wouldn't normally be able to see.

Reply:
Isn't `ephRequest.UserID` the one who would actually be receiving the post though? If I understand the code correctly, the first two IDs are the one making the ephemeral post and the third is the one receiving it, so I'd expect it to be done based on the recipient's user ID.

Reply:
Maybe... I'm a bit confused by the difference between `PostEphemeral.UserID` and `PostEphemeral.Post.UserId` (which is set to `c.AppContext.Session().UserId`).

Reply:
Recapping what we discussed: it looks like we have the correct user ID here.